src/pretix/presale/forms/checkout.py

```diff
@@ -295,7 +301,28 @@ def __init__(self, *args, **kwargs):
                     label=q.question, required=q.required,
                     help_text=q.help_text,
                     initial=initial.file if initial else None,
-                    widget=UploadedFileWidget(position=pos, event=event, answer=initial)
+                    widget=UploadedFileWidget(position=pos, event=event, answer=initial),
+                )
+            elif q.type == Question.TYPE_DATE:
+                field = forms.DateField(
+                    label=q.question, required=q.required,
+                    help_text=q.help_text,
+                    initial=dateutil.parser.parse(initial.answer).date() if initial and initial.answer else None,
+                    widget=DatePickerWidget(),
+                )
+            elif q.type == Question.TYPE_TIME:
+                field = forms.TimeField(
+                    label=q.question, required=q.required,
+                    help_text=q.help_text,
+                    initial=dateutil.parser.parse(initial.answer).astimezone(tz).time() if initial and initial.answer else None,
+                    widget=TimePickerWidget(time_format=get_format('TIME_INPUT_FORMATS')[2]),
+                )
+            elif q.type == Question.TYPE_DATETIME:
+                field = forms.SplitDateTimeField(
+                    label=q.question, required=q.required,
+                    help_text=q.help_text,
+                    initial=dateutil.parser.parse(initial.answer).astimezone(tz) if initial and initial.answer else None,
+                    widget=SplitDateTimePickerWidget(time_format=get_format('TIME_INPUT_FORMATS')[2]),
```

Comment:
I've gone through all of Django's supported locales and this will break if we ever add:
* Esperanto
* Hungarian

Reply:
Ok. sorry for not thinking about that. Will read up on that.

Reply:
I created a `get_format_without_seconds` function in `pretix/control/utils/i18n.py` (and also one for javascript formats) for that and will probably add a test. Maybe this should be moved to `pretix/base`, since it's in `pretix/presale` now?

Reply:
Yes, you can move it, if you want, it makes sense.